src/ai/defaultai.cc

```diff
@@ -4613,18 +4782,16 @@ bool DefaultAI::check_productionsites(const Time& gametime) {
 			return false;
 		}
 
-		const uint32_t remaining_trees = map.find_immovables(
-		   game(),
-		   Widelands::Area<Widelands::FCoords>(map.get_fcoords(site.site->get_position()), radius),
-		   nullptr,
-		   Widelands::FindImmovableAttribute(Widelands::MapObjectDescr::get_attribute_id("tree")));
-
 		if (site.site->get_statistics_percent() >
 		    std::abs(management_data.get_military_number_at(117)) / 2) {
 			return false;
 		}
 
-		if (remaining_trees > trees_nearby_treshold_ / 3) {
+		unsigned trees_nearby = find_immovables_nearby(site.site->descr().collected_attributes(),
```

Comment:
this does not deliver the same number as the old code. But so far I don't know why

Reply:
Well here I was wrong. After inserting some debugging logs I confirm this line is working.